src/System.Buffers/src/System/Buffers/BufferPool.cs

```diff
@@ -41,7 +36,8 @@ public T[] RentBuffer(int size, bool clearBuffer = false)
             }
             else
             {
-                buffer = _buckets[SelectBucketIndex(size)].Rent();
+                int index = SelectBucketIndex(size);
+                buffer = LazyInitializer.EnsureInitialized(ref _buckets[index], () => new BufferBucket<T>(GetMaxSizeForBucket(index), NUMBER_OF_BUFFERS_IN_BUCKET)).Rent();
```

Comment:
Written this way it's actually two allocations: the delegate and the closure.  If it could be instead written in a way where the delegate didn't need to close over anything, the closure allocation would go away and the compiler would cache the delegate.